src/compiler/glue-compiler.cc

```diff
@@ -101,6 +101,36 @@ static hilti::ID extract_id(const std::string& chunk, size_t* i) {
     return hilti::ID(hilti::util::replace(id, "%", "0x25_"));
 }
 
+static hilti::Type extract_type(const std::string& chunk, size_t* i) {
+    eat_spaces(chunk, i);
+
+    // We currently only parse Spicy types that can appear in parameters of
+    // built-in hooks--which are not many.
+    auto token = extract_id(chunk, i);
+
+    if ( token == hilti::ID("string") )
+        return hilti::type::String();
+
+    if ( token == hilti::ID("uint64") )
+        return hilti::type::UnsignedInteger(64);
+
+    throw ParseError("mismatching type");
+}
+
+static hilti::type::function::Parameter extract_parameter(const std::string& chunk, size_t* i) {
+    eat_spaces(chunk, i);
+
+    auto id = extract_id(chunk, i);
+
+    if ( ! looking_at(chunk, *i, ":") )
+        throw ParseError("expected ':'");
+
+    eat_token(chunk, i, ":");
+
+    auto type = extract_type(chunk, i);
+    return builder::parameter(id, type);
```

Comment:
Nit: We could move the parameters here,

```suggestion
    return builder::parameter(std::move(id), std::move(type));
```